tests/test.hs

```diff
@@ -34,6 +34,12 @@ tests = testGroup "Data.Yaml.Combinators"
   , testCase "Expect an object, get an object" $
       runParser (object $ field "foo" number) (Object [("foo", Number 1)]) @?=
         Right 1
+  , testCase "Validated String, accepts anything" $
+      runParser (validate string Right) (String "foo") @?=
+        Right "foo"
+  , testCase "Validated String, rejects everything" $
+      runParser (validate string (const $ Left "contrarianism")) (String "foo") @?=
+        Left (ParseError 1 (ExpectedAsPartOf "contrarianism" (String "foo")))
```

Comment:
This error doesn't quite make sense; I think `ExpectedInsteadOf` would be a better fit here.